src/structures/MessageEmbed.js

```diff
@@ -345,15 +345,32 @@ class MessageEmbed {
     return this;
   }
 
+  /**
+   * The options to provide for setting an author for a {@link MessageEmbed}.
+   * @typedef {Object} EmbedAuthorData
+   * @property {string} name The name of this author.
+   * @property {string} [url] The URL of this author.
+   * @property {string} [iconURL] The icon URL of this author.
+   */
+
+  // TODO: Remove the deprecated code in the following method and typings.
   /**
    * Sets the author of this embed.
-   * @param {string} name The name of the author
-   * @param {string} [iconURL] The icon URL of the author
-   * @param {string} [url] The URL of the author
+   * @param {string|EmbedAuthorData} options The options to provide for the author.
```

Comment:
Technically options is optional now
```suggestion
   * @param {string|EmbedAuthorData} [options={}] The options to provide for the author.
```

Reply:
This has given me a thought: how does one remove an author with this method? Should we allow `null` to empty the author object just like`MessageEmbed#setTimestamp()`?

Reply:
I mean calling .setAuthor will already make it empty, no?

Reply:
That's true, but calling a set method with no arguments feels like it should error or be set with a default value. `null` is being explicit to remove. Thoughts?

Reply:
We don't do it for any other method, like setDescription or such, they all require a string, be it empty... @discordjs/the-big-4 what do?

Reply:
Currently, it won't set to `null`, but rather will cause an error to be thrown on line 372:

```js
    const { name, url, iconURL } = options;
```

Perhaps we can do a manual check and set it to the default value when `null` is passed?

Reply:
I'd be fine with clearing the underlying properties (or setting to null) when null is passed to a `set__` method.